dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/services/tasks/Ds3GetJob.java

```diff
@@ -135,14 +138,15 @@ private void transferFromSelectedItem(final String bucketName,
         final String prefix = getParent(selectedItem.getFullName(), BP_DELIMITER);
         final FluentIterable<Ds3Object> ds3Objects = getDS3Objects(bucketName, selectedItem);
         final long totalJobSize = getTotalJobSize(ds3Objects);
-        final Ds3ClientHelpers.Job job;
         if (ds3Objects.isEmpty()) {
             LOG.info("Did not create job because items were empty");
             return;
         }
         ds3Objects.filter(ds3 -> isEmptyDirectory(ds3, BP_DELIMITER)).forEach(ds3Object -> Ds3GetJob.buildEmptyDirectories(ds3Object, fileTreePath, loggingService));
         try {
-            job = getJobFromIterator(bucketName, ds3Objects.filter(ds3Object -> !isEmptyDirectory(ds3Object, delimiter)));
+            if (job == null) {
```

Comment:
Instead of checking for nullability here, either extract this logic into it's own method, or split this method into smaller pieces that each does one thing, like preparing the list of work, and then the creation of the job.

Splitting it out either way, should help isolate the side effect for the job creation.